tss-esapi/src/ffi/data_zeroize.rs

```diff
@@ -0,0 +1,444 @@
+                    self.details.ecdsa.ffi_data_zeroize();
+                },
+                EccSchemeAlgorithm::EcDh => unsafe {
+                    self.details.ecdh.ffi_data_zeroize();
+                },
+                EccSchemeAlgorithm::EcDaa => unsafe {
+                    self.details.ecdaa.ffi_data_zeroize();
+                },
+                EccSchemeAlgorithm::Sm2 => unsafe {
+                    self.details.sm2.ffi_data_zeroize();
+                },
+                EccSchemeAlgorithm::EcSchnorr => unsafe {
+                    self.details.ecschnorr.ffi_data_zeroize();
+                },
+                EccSchemeAlgorithm::EcMqv => unsafe {
+                    self.details.ecmqv.ffi_data_zeroize();
+                },
+                EccSchemeAlgorithm::Null => {}
+            }
+        }
+        self.scheme.zeroize();
+    }
+}
+
+impl FfiDataZeroize for TPMT_KDF_SCHEME {
+    fn ffi_data_zeroize(&mut self) {
+        if let Ok(key_derivation_function) = KeyDerivationFunction::try_from(self.scheme) {
+            match key_derivation_function {
+                KeyDerivationFunction::Kdf1Sp800_56a => unsafe {
+                    self.details.kdf1_sp800_56a.ffi_data_zeroize();
+                },
+                KeyDerivationFunction::Kdf2 => unsafe {
+                    self.details.kdf2.ffi_data_zeroize();
+                },
+                KeyDerivationFunction::Kdf1Sp800_108 => unsafe {
+                    self.details.kdf1_sp800_108.ffi_data_zeroize();
+                },
+                KeyDerivationFunction::Mgf1 => unsafe {
+                    self.details.mgf1.ffi_data_zeroize();
+                },
+                KeyDerivationFunction::Null => {}
+            }
+        }
+        self.scheme.zeroize();
+    }
+}
+// Tickets
+implement_ffi_data_zeroizer_trait_for_ticket_type!(TPMT_TK_CREATION);
+implement_ffi_data_zeroizer_trait_for_ticket_type!(TPMT_TK_VERIFIED);
+implement_ffi_data_zeroizer_trait_for_ticket_type!(TPMT_TK_AUTH);
+implement_ffi_data_zeroizer_trait_for_ticket_type!(TPMT_TK_HASHCHECK);
+
+///////////////////////////////////////////////////////////////////////////
+/// UNIT TESTS FOR FFI DATA ZEROIZE
+///////////////////////////////////////////////////////////////////////////
+macro_rules! implement_zeroize_test_for_named_field_structured_buffer_type {
+    ($tss_type:ident, $buffer_field_name:ident, $fn_name:ident) => {
+        #[test]
+        fn $fn_name() {
+            const fn fill_with_ones<const SIZE: usize>() -> [u8; SIZE] {
+                [1u8; SIZE]
+            }
+
+            let expected: $tss_type = unsafe { std::mem::zeroed() };
+            let mut actual = $tss_type {
+                size: 10,
+                $buffer_field_name: fill_with_ones(),
+            };
+            actual.ffi_data_zeroize();
+            assert_eq!(expected.size, actual.size);
+            assert_eq!(expected.$buffer_field_name, actual.$buffer_field_name);
+        }
+    };
+}
```

Comment:
Excellent, thanks!

Reply:
Yes, I will add that when I have the time. Probably sometime next year.